test/e2e/e2e.go

```diff
@@ -46,3 +48,25 @@ func CreateRouteAndConfig(t *testing.T, clients *test.Clients, image string, opt
 	_, err := test.CreateRoute(t, clients, names)
 	return names, err
 }
+
+// WaitForScaleToZero will wait for the deployment specified by names to scale
+// to 0 replicas. Will wait up to 3 minutes before failing.
+func WaitForScaleToZero(t *testing.T, names test.ResourceNames, clients *test.Clients) {
+	t.Helper()
+	deploymentName := names.Revision + "-deployment"
+	t.Logf("Waiting for %q to scale to zero", deploymentName)
+	err := pkgTest.WaitForDeploymentState(
+		clients.KubeClient,
+		deploymentName,
+		func(d *v1beta1.Deployment) (bool, error) {
+			t.Logf("Deployment %q has %d replicas", deploymentName, d.Status.ReadyReplicas)
+			return d.Status.ReadyReplicas == 0, nil
+		},
+		"DeploymentIsScaledDown",
+		test.ServingNamespace,
+		3*time.Minute,
```

Comment:
Requires another API call, but would be nice if this was a multiple of the configured scale to zero threshold rather than an arbitrary time. https://github.com/knative/serving/blob/master/config/config-autoscaler.yaml#L85